DataDog/datadog-operator
# [CECO-2][DatadogMonitor] Configure force sync period

### What does this PR do?

* Allows the user to configure the force sync period for API monitors (version stored in Datadog back-end) using `DD_MONITOR_FORCE_SYNC_PERIOD` env var

### Motivation

* https://datadoghq.atlassian.net/browse/CECO-2 : feature request

### Describe your test plan

1. Deploy this version of the operator with the env var `DD_MONITOR_FORCE_SYNC_PERIOD` set to `5`
2. Deploy an example monitor, e.g. `kubectl apply -f examples/datadogmonitor/metric-monitor-imagepullbackoff.yaml`
3. After a minute or so, in the Datadog UI, edit the monitor query, e.g. changing the threshold to 2 and the 10 minutes to 5 minutes instead (creating a drift from the `DatadogMonitor`)
4. Ensure the monitor is reverted back to the `DatadogMonitor` query within 5 minutes at most from your modification : you can identify this by looking at the monitor audit trail in the UI or in kubernetes by reviewing the `Monitor Last Force Sync Time` in the `DatadogMonitor` (within `Status`) or the associated events
    <img width="2373" alt="image" src="https://github.com/user-attachments/assets/5647109b-453e-4ac9-b3cc-39b0a45c9295">
5. Modify `DD_MONITOR_FORCE_SYNC_PERIOD` to a non integer value, e.g. "0,5" and assess the presence of the operator log complaining the value is invalid: 
```json
{"level":"ERROR","ts":"2024-09-11T12:44:03.864Z","logger":"controllers.DatadogMonitor","msg":"Invalid value for monitor force sync period. Defaulting to 60 minutes.","datadogmonitor":{"name":"pods-imagepullbackoff","namespace":"system"},"error":"strconv.Atoi: parsing \"0,5\": invalid syntax"}
```


### Checklist
